### PR TITLE
🌐 Web Chat Channel

### PR DESCRIPTION
**Pull request by paulobernardoaf**
Integration with Weni's Web Chat, a browser Chat that can be embedded on a website and connected with RapidPro.
[Weni Web Chat Widget](https://github.com/Ilhasoft/weni-webchannel)
[Weni Web Chat Socket](https://github.com/Ilhasoft/weni-webchat-socket)

Courier Handler: https://github.com/nyaruka/courier/pull/372

To run and test the stack locally you will need to run the Socket server and a program like [ngrok](https://ngrok.com/) to be able to generate a public URL for it and both RapidPro and Web Chat widget use it.

The Socket server requires an S3 configuration if you wish to be able to send attachments from the widget to RapidPro.

Required environment variables
```bash
export WWC_LOG_LEVEL="trace" WWC_PORT=8081 WWC_S3_ENDPOINT=https://s3.sa-east-1.amazonaws.com WWC_S3_REGION=sa-east-1 WWC_S3_BUCKET=your-bucket-name WWC_S3_ACCESS_KEY=your_s3_access_key WWC_S3_SECRET_KEY=your_s3_secret_key
```

To run the Socket server
```go
go run ./api
```

The widget is currently hosted on google cloud, and the following HTML page is able to use it.

```html
<html>
  <head>
    <meta  name="viewport"  content="width=device-width, initial-scale=1">
    <script src="https://storage.googleapis.com/push-webchat/wwc-latest.js"></script>
  </head>
  <body>
    <div id="webchat"/>

    <script>
      WebChat.default.init({
        selector: "#webchat",
        initPayload: "Hello",
        channelUuid: '11111111-2222-3333-4444-555555555555', // External Channel UUID from Push
        host: 'https://app.rapidpro.io', // Host URL from RapidPro
        socketUrl: "https://socket-dev.weni.ai", // The Socket Url
        title: "Title",
        subtitle: "Subtitle",
        params: {
          images: {
            dims: {
              width: 300,
              height: 200
            }
          },
          storage: "session"
        },
      })
    </script>
  </body>
</html>
```

### CODE REVIEW

File: templates/channels/types/weniwebchat/claim.haml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+-extends "channels/channel_claim_form.html"
+-load i18n compress temba
+
+-block pre-form
+    -blocktrans with name=brand.name trimmed
```
Comment (rowanseymour) [inline templates/channels/types/weniwebchat/claim.haml:5]: indentation should be 2

File: templates/channels/types/weniwebchat/claim.haml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+-extends "channels/channel_claim_form.html"
+-load i18n compress temba
```
Comment (rowanseymour) [inline templates/channels/types/weniwebchat/claim.haml:2]: don't think `compress` and `temba` are needed

File: temba/channels/types/weniwebchat/type.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
+from django.utils.translation import ugettext_lazy as _
+
+from temba.contacts.models import URN
+
+from ...models import ChannelType
+from .views import ClaimView
+
+CONFIG_BASE_URL = "base_url"
```
Comment (norkans7) [inline temba/channels/types/weniwebchat/type.py:8]: use `Channel.CONFIG_BASE_URL` here

File: temba/channels/types/weniwebchat/views.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+        user = self.request.user
+        org = user.get_org()
+
+        data = form.cleaned_data
+
+        name = form.cleaned_data["name"]
+        base_url = data["base_url"]
+
+        config = {CONFIG_BASE_URL: base_url}
```
Comment (norkans7) [inline temba/channels/types/weniwebchat/views.py:32]: `Channel.CONFIG_BASE_URL`

File: temba/channels/types/weniwebchat/views.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+        self.object = Channel.create(
+            org, self.request.user, None, self.channel_type, config=config, name=name, address=name
```
Comment (norkans7) [inline temba/channels/types/weniwebchat/views.py:35]: Address is the same as the name?
is that really required when we send?
I think address can be "" if not required.

**Pull request closed without merging**